### ISSUE TITLE
As a 3P dev, I need to have a reference entry for the new error type `result-size-exceeded`

### ISSUE THREAD

**Issue by kebetsi** [pryv/dev-site#14]
The API version 1.1.7 has a new error type when requesting too much events in a batch or websockets call, it must be referenced in the API.

**Comment by sgoumaz** [comment]
I didn't notice it in the code so commenting here: why name the error `result-size-exceeded`? How a "result size" can be "exceeded" doesn't make sense to me; I'd propose `too-many-results` instead. One can call that nitpicking, but sensible naming is so important especially addressing an outside audience. Really. Trust me.

**Comment by kebetsi** [comment]
Fine for me, thanks for the input @sgoumaz. Can you take a look at the message that is outputed with this error, please?

**Comment by sgoumaz** [comment]
For the message, I'd rephrase it to something like `Your request gave too many results (the limit is ${limit}). You could narrow the scope or use paging.` (not sure what "direct API call" meant)

BTW:

- The [error id](https://github.com/pryv/service-core/blob/62262fe3dd75e02914c1bf39dad33acd80b58f89/components/errors/src/ErrorIds.js#L27) should use the same PascalCase convention as the others
- Both the error id and the [factory function](https://github.com/pryv/service-core/blob/62262fe3dd75e02914c1bf39dad33acd80b58f89/components/errors/src/factory.js#L139) declarations should be ordered alphabetically like the others (alphabetical order used here as it's a list of definitions)

**Comment by kebetsi** [comment]
The information that I wanted to transmit was:
- the limit size
- to recommend using the HTTP call GET /events (that's what I meant by "direct API call") when querying big amounts, instead of in batch or socket.io.  

Other comments:
- I saw the errorId PascalCase issue, will fix it.
- I didn't notice the alphabetic ordering, will sort this out
- I had a question about [this comment](https://github.com/pryv/service-core/blob/62262fe3dd75e02914c1bf39dad33acd80b58f89/components/errors/src/ErrorIds.js#L11) which seems out of place since these are general API call errors and not socket.io specific. My proposition is to remove this comment.

**Comment by sgoumaz** [comment]
Then maybe `Your request gave too many results (the limit is ${limit}). Directly calling the API method (i.e. not batching calls), narrowing request scope or paging can help.`

That comment is there because without it readers may be left wondering how such an error could occur (the other errors are hopefully self-explanatory). I wouldn't remove it at least for the reason you mention – this is a list of the errors the API can throw, for both HTTP and websocket calls, and it's pretty normal that some are specific to either mode (e.g. `MissingHeader` is obviously HTTP-only).

**Comment by kebetsi** [comment]
I have implemented the aforementioned modifications in `service-core` and updated the API version in api.pryv.com.
I was going to add the new error to the API reference, but I noticed that many errors (such as `periodsOverlap`, `itemAlreadyExists`, `invalidMethod`, etc.) weren't referenced there. Is this an oversight or is this on purpose? @sgoumaz

**Comment by sgoumaz** [comment]
Usual errors (common to most/many methods) are [listed as such](https://api.pryv.com/reference/#usual-errors). Errors specific to particular methods are described along with the methods ([example for `events.create`](https://api.pryv.com/reference/#create-event)). A quick search would've shown that.

But thanks anyway because `invalid-method` (thrown when given an invalid method id over websockets) is indeed undocumented, and probably should be. If you agree I think a simple mention of it in [the relevant subsection](https://api.pryv.com/reference/#calling-methods) is appropriate.

For `too-many-results`, I would add it to the "usual errors" section but precising that it can only happen when querying data (I suppose only `.get` and batch calls).
